app/web/features/auth/useAuthStore.ts

```diff
@@ -131,12 +135,53 @@ export default function useAuthStore() {
         }
         setLoading(false);
       },
+      async checkAuthStatus() {
+        try {
+          const res = await service.user.getAuthState();
+          if (res.loggedIn) {
+            console.log(
+              "We thought we were not logged in but an API call shows we were."
+            );
+            const auth = res.authRes!;
+            setUserId(auth.userId);
+            Sentry.setUser({ id: auth.userId.toString() });
+
+            //this must come after setting the userId, because calling setQueryData
+            //will also cause that query to be background fetched, and it needs
+            //userId to be set.
+            setJailed(auth.jailed);
+            setAuthenticated(true);
+          }
```

Comment:
Would you want to logout here if `res.loggedIn` equals false? In an else clause for example?